src/App.tsx

```diff
@@ -15,36 +17,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  name = 'name',
+  length = 'length',
+}
+
+interface Filter {
+  field: SortField;
+  isreverse: boolean;
+}
+
+const preparedGoods = (
+  goods: string[],
+  { field, isreverse }: Filter,
+): string[] => {
+  const sortedGoods = [...goods];
+
+  if (!field) {
+    return isreverse ? sortedGoods.reverse() : sortedGoods;
+  }
+
+  sortedGoods.sort((good1, good2) => {
+    switch (field) {
+      case SortField.name:
+        return good1.localeCompare(good2);
+      case SortField.length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  return isreverse ? sortedGoods.reverse() : sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortField | ''>('');
```

Comment:
The `sortField` state is initialized with an empty string, which is not compatible with the `SortField` enum type. Consider changing the type to `SortField | null` and initializing it with `null` instead of an empty string.